specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2024-11-01/AppServiceCertificateOrders.json

```diff
@@ -0,0 +1,1801 @@
+              "type": "string",
+              "readOnly": true
+            },
+            "contact": {
+              "$ref": "#/definitions/CertificateOrderContact",
+              "description": "Contact info",
+              "readOnly": true
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AppServiceCertificatePatchResource": {
+      "description": "Key Vault container ARM resource for a certificate that is purchased through Azure.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../Microsoft.Web/stable/2024-11-01/CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AppServiceCertificate",
+          "description": "Core resource properties",
+          "type": "object",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AppServiceCertificateResource": {
+      "description": "Key Vault container ARM resource for a certificate that is purchased through Azure.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../Microsoft.Web/stable/2024-11-01/CommonDefinitions.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AppServiceCertificate",
+          "description": "Core resource properties",
+          "type": "object",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "CertificateDetails": {
+      "description": "SSL certificate details.",
+      "type": "object",
+      "properties": {
+        "version": {
+          "format": "int32",
+          "description": "Certificate Version.",
+          "type": "integer",
+          "readOnly": true
+        },
+        "serialNumber": {
+          "description": "Certificate Serial Number.",
+          "type": "string",
+          "readOnly": true
+        },
+        "thumbprint": {
+          "description": "Certificate Thumbprint.",
+          "type": "string",
+          "readOnly": true
+        },
+        "subject": {
+          "description": "Certificate Subject.",
+          "type": "string",
+          "readOnly": true
+        },
+        "notBefore": {
+          "format": "date-time",
+          "description": "Date Certificate is valid from.",
+          "type": "string",
+          "readOnly": true
+        },
+        "notAfter": {
+          "format": "date-time",
+          "description": "Date Certificate is valid to.",
+          "type": "string",
+          "readOnly": true
+        },
+        "signatureAlgorithm": {
+          "description": "Certificate Signature algorithm.",
+          "type": "string",
+          "readOnly": true
+        },
+        "issuer": {
+          "description": "Certificate Issuer.",
+          "type": "string",
+          "readOnly": true
+        },
+        "rawData": {
```

Comment:
>rawData [](http://example.com/codeflow?start=9&length=7)

Include information about the encoding method in the description. Also, if this is a secret it needs the x-ms-secret extension and must be scrubbed from GET payloads. If clients need to be able to read secrets, it must be via separate POST action to allow for separate RBAC. This definition is used many places, but the property can only be populated in PUT, PATCH, or POST methods.